src/Features/SkySync.cpp

```diff
@@ -0,0 +1,441 @@
+﻿#include "SkySync.h"
+
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT(
+	SkySync::Settings,
+	Enabled,
+	UseAlternateSunPath,
+	MoonLightSource)
+
+void SkySync::DrawSettings()
+{
+	ImGui::Checkbox("Enabled", &settings.Enabled);
+	ImGui::Checkbox("Use alternate sun path", &settings.UseAlternateSunPath);
+	ImGui::SliderInt("Moon light source", &settings.MoonLightSource, 0, static_cast<uint8_t>(MoonLightSource::Count) - 1, MoonLightSourceNames[settings.MoonLightSource]);
+}
+
+void SkySync::LoadSettings(json& o_json)
+{
+	settings = o_json;
+}
+
+void SkySync::SaveSettings(json& o_json)
+{
+	o_json = settings;
+}
+
+void SkySync::RestoreDefaultSettings()
+{
+	settings = {};
+}
+
+void SkySync::DataLoaded()
+{
+	moonAndStarsLoaded = GetModuleHandle(L"po3_MoonMod.dll");
+	if (moonAndStarsLoaded)
+		logger::info("[Sky Sync] Moon and Stars detected, compatibility enabled");
+
+	auto disableDueToConflict = [&](std::string_view conflictName) {
+		failedLoadedMessage = fmt::format("Sky Sync has been disabled as {} has been detected, both cannot be used together", conflictName);
+		loaded = false;
+		logger::warn("{}", failedLoadedMessage);
+	};
+
+	const auto data = RE::TESDataHandler::GetSingleton();
+	if (data && (data->LookupLoadedModByName("DVLaSS.esp"sv) || data->LookupLoadedLightModByName("DVLaSS.esp"sv)))
+		disableDueToConflict("DVLaSS");
+	else if (GetModuleHandle(L"EVLaS.dll"))
+		disableDueToConflict("EVLaS");
+}
+
+void SkySync::PostPostLoad()
```

Comment:
Hooking this now means that it doesn't actually disable.

Reply:
I've modified the logic in latest so that conflict detection for DLLs are now checked in PostPostLoad then hooks are applied after. ESP conflicts are checked in DataLoaded after mods have loaded - the hooks will already be in place at this point but the module will self-disable and show up as an issue in the UI.

This should fix the intermittent crashing for some users when hooks are applied in DataLoaded.